test/batchReceiver.spec.ts

```diff
@@ -421,4 +423,533 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
+  });
+
+  async function deferMsgAndGetSequenceNum(
+    client: QueueClient | SubscriptionClient
+  ): Promise<Long> {
+    const receivedMsgs = await client.receiveBatch(1);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    if (!receivedMsgs[0].sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    const sequenceNumber = receivedMsgs[0].sequenceNumber;
+    await receivedMsgs[0].defer();
+    return sequenceNumber;
+  }
+
+  it("Abandon a message received from recevieDefferedMessage using queue", async function(): Promise<
+    void
+  > {
+    await queueClient.send(testMessages[0]);
+    const sequenceNumber = await deferMsgAndGetSequenceNum(queueClient);
+
+    const deferredMsgs = await queueClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.abandon();
+
+    await testPeekMsgsLength(queueClient, 1);
+
+    if (!sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    const abandonedMsgs = await queueClient.receiveDeferredMessage(sequenceNumber);
+    if (!abandonedMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(abandonedMsgs.body, testMessages[0].body);
+    should.equal(abandonedMsgs.messageId, testMessages[0].messageId);
+
+    await abandonedMsgs.complete();
+
+    await testPeekMsgsLength(queueClient, 0);
+  });
+
+  it("Abandon a message received from recevieDefferedMessage using subscription", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    const sequenceNumber = await deferMsgAndGetSequenceNum(subscriptionClient);
+
+    const deferredMsgs = await subscriptionClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.abandon();
+
+    await testPeekMsgsLength(subscriptionClient, 1);
+
+    if (!sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    const abandonedMsgs = await subscriptionClient.receiveDeferredMessage(sequenceNumber);
+    if (!abandonedMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(abandonedMsgs.body, testMessages[0].body);
+    should.equal(abandonedMsgs.messageId, testMessages[0].messageId);
+
+    await abandonedMsgs.complete();
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("Dead letter a message received from recevieDefferedMessage using queue", async function(): Promise<
```

Comment:
Change this to `Deadlettering a deferred message moves it to dead letter queue`.
Same for subscription